### PR TITLE
feat(annotations): Show versioned annotation activity cards

### PR DESCRIPTION
**Pull request by ConradJChan**
Propagate the current file_version down so the `AnnotationActivityItem` is aware of whether the annotation is on the current file version or not

TODO:
 - [x] unit tests

### CODE REVIEW

File: src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js [hunk 1]
```diff
@@ -79,6 +81,7 @@ const ActiveState = ({
             {items.map((item: FeedItem) => {
                 const isFocused = item === activeEntry;
                 const refValue = isFocused ? activeFeedItemRef : undefined;
+                const fileVersionId = getProp(item, 'file_version.id');
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js:84]: Maybe `itemFileVersionId` to be somewhat more clear that this is activity item-specific rather than file-specific?

File: src/elements/content-sidebar/activity-feed/activity-feed/__tests__/ActiveState.test.js [hunk 2]
```diff
@@ -148,4 +153,28 @@ describe('elements/content-sidebar/ActiveState/activity-feed/ActiveState', () =>
...
     });
+
+    test.each`
+        currentFileVersionId | isCurrentVersion
+        ${'123'}             | ${true}
+        ${'456'}             | ${false}
+    `(
+        'should correctly reflect annotation activity isCurrentVersion as $isCurrentVersion based on file version id as $currentFileVersionId',
+        ({ currentFileVersionId, isCurrentVersion }) => {
+            const wrapper = shallow(
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/activity-feed/__tests__/ActiveState.test.js:164]: I know this is an existing test file, but would a `getWrapper` helper be worthwhile to avoid redeclaring all the base props for each test?

File: src/elements/content-sidebar/ActivitySidebar.js [hunk 3]
```diff
@@ -144,12 +145,15 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
         const { requestId } = meta || {};
         const { currentUser } = this.state;
         const isPending = action === 'create_start';
+        const currentFileVersionId = getProp(file, 'file_version.id');
+        // Patch the pending annotation with the current file version id
+        const patchedAnnotation = { ...annotation, file_version: { id: currentFileVersionId } };
```
Comment (jstoffan) [inline src/elements/content-sidebar/ActivitySidebar.js:150]: Can users create annotations on older versions?

Comment (ConradJChan) [reply src/elements/content-sidebar/ActivitySidebar.js]: Good callout, patching with the current version won't work. I'll make the change upstream in box-annotations

File: src/elements/content-sidebar/activity-feed/annotations/__tests__/AnnotationActivity.test.js [hunk 4]
```diff
@@ -25,8 +27,13 @@ describe('elements/content-sidebar/ActivityFeed/annotations/AnnotationActivity',
         created_by: { name: 'Jane Doe', id: 10 },
         currentUser,
         description: { message: 'test' },
+        file_version: {
+            id: '456',
+            version_number: '2',
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/annotations/__tests__/AnnotationActivity.test.js:32]: This field won't be present on a "new" annotation. Is that okay?

Comment (ConradJChan) [reply src/elements/content-sidebar/activity-feed/annotations/__tests__/AnnotationActivity.test.js:32]: I think that should be okay for now. It's a corner case that might come up with real-time updates but we're not there yet
